cmd/osm-controller/osm-controller.go

```diff
@@ -245,22 +246,34 @@ func main() {
 		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error initializing ADS server")
 	}
 
-	// Initialize function and HTTP health probes
-	funcProbes := []health.Probes{xdsServer}
-	httpProbes := getHTTPHealthProbes()
-
-	// Initialize the http server and start it
-	httpServer := httpserver.NewHTTPServer(funcProbes, httpProbes, metricsstore.DefaultMetricsStore, constants.MetricsServerPort)
-	httpServer.Start()
-
 	if err := createControllerManagerForOSMResources(certManager); err != nil {
 		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error creating controller manager to reconcile OSM resources")
 	}
 
-	// Expose /debug endpoints and data only if the enableDebugServer flag is enabled
+	// Initialize OSM's http service server
+	httpServer := httpserver.NewHTTPServer(constants.OSMServicePort)
+
+	// Health/Liveness probes
+	funcProbes := []health.Probes{xdsServer}
+	httpServer.AddHandlers(map[string]http.Handler{
+		"/health/ready": health.ReadinessHandler(funcProbes, getHTTPHealthProbes()),
+		"/health/alive": health.LivenessHandler(funcProbes, getHTTPHealthProbes()),
+	})
+	// Metrics
+	httpServer.AddHandler("/metrics", metricsstore.DefaultMetricsStore.Handler())
+	// Version
+	httpServer.AddHandler("/version", version.GetVersionHandler())
```

Comment:
Can we move this to a separate func `initializeHTTPServerHandlers()` or something to keep `main()` concise.

Reply:
I originally did this but, but passing xdsServer was kind of ugly.  I can do it again though.

Reply:
this can be fixed later as well.